packages/@ember/-internals/runtime/lib/mixins/array.js

```diff
@@ -491,6 +491,49 @@ const ArrayMixin = Mixin.create(Enumerable, {
     invalidate any related properties. Pass the starting index of the change
     as well as a delta of the amounts to change.
 
+    ```javascript
+    const Post = EmberObject.extend({
+      body: '',
+      save() {}
+    })
+
+    export default Component.extend({
+
+      attemptsToModify: 0,
+      successfulModifications: 0,
+      posts: null,
+
+      init() {
+        this._super(...arguments);
+        this.posts = [1, 2, 3].map(i => Post.create({ body: i }));
+        this.posts.addArrayObserver(this, {
+          willChange() {
+            this.incrementProperty('attemptsToModify');
+          },
+          didChange() {
+            this.incrementProperty('successfulModifications');
+          }
+        });
+      },
+
+      actions: {
+        editPost(post, newContent) {
+          let oldContent = post.body;
+          this.posts.arrayContentWillChange(); // attemptsToModify = 1
```

Comment:
You are not passing any of the two required params to `arrayContentWillChange`, can you update?